README.md

```diff
@@ -43,3 +43,68 @@ make ghcid-test
 ---
 
 [![Build Status](https://travis-ci.org/input-output-hk/fm-ledger-rules.svg?branch=master)](https://travis-ci.org/input-output-hk/fm-ledger-rules)
+
+
+# nix-build Infrastructure
+
+The artifacts in this repository can be built and tested using nix. This is
+additionally used by the Hydra CI to test building, including cross-compilation
+for other systems.
+
+## To add a new Haskell project
+
+To add a new Haskell project, you should do the following:
+
+1. Create the project in the usual way. It should have an appropriate `.cabal` file.
+2. Add the project to the [top-level stack.yaml](./stack.yaml), configuring
+   dependencies etc as needed. If your project's configuration deviates too far
+   from the [snapshot in
+   ``cardano-prelude`](https://github.com/input-output-hk/cardano-prelude/blob/master/snapshot.yaml),
+   then you may have to submit a PR there to update that snapshot.
+3. At this point, test that your new project builds using `stack build <project_name>`.
+4. Run the [nix-tools-generate](./scripts/nix-tools-generate.sh) script to
+   rebuild the nix configuration from your stack.yaml file.
+5. Test that you can build your new project by running the following: `nix build
+   -f default.nix nix-tools.libs.<project_name>`. If you have executables, then
+   you may also try building these using the `nix-tools.exes.<executable_name>`
+   attribute path. A good way to see what's available is to execute `:l
+   default.nix` in `nix repl`. This will allow you to explore the potential
+   attribute names.
+6. If your project uses template haskell, it may not build in the previous step.
+   In this case, add it to the `th-packages` list in [pkgs.nix](./nix/pkgs.nix).
```

Comment:
`th-packages` should only strictly be necessary to provide the build infrastructure with enough information for cross-compilation.